### DISCUSSION TITLE
Security question

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#2875] [Q&A]
### Question

Hello,

How secure is NiceGUI library against malicious user? For example, if an input and a button are disabled in Python code, but a user launches browser Developer Tools and manually removes "disabled" attribute from the corresponding controls in the web page, then clicks the button, would Python `on_click` handler be executed?

**Comment by falkoschindler** [comment]
Hi @me21,

This specific issue has been addressed in #983 and PR #1030 (version 1.2.21).
But it's always good to keep this attack vector in mind. Maybe there are other situations we haven't taken care of.

**Comment by me21** [comment]
Regarding that PR, I have a question:

For example, `ui.button` is a descendant of both `Visibility` and `DisableableElement`. How would its `is_ignoring_events` property work in that case? Would only one version be called, ignoring the other, or both property versions will be ORed together?

**Comment by falkoschindler** [comment]
I guess they are ORed:

https://github.com/zauberzeug/nicegui/blob/94cc62c8317360257651458a8e452e746c55261c/nicegui/elements/mixins/disableable_element.py#L18-L23

But if in doubt, we should try it out or write a pytest.
